diff --git a/platform/srf06-cc26xx/contiki-conf.h b/platform/srf06-cc26xx/contiki-conf.h
index 382b572..c3a4109 100644
--- a/platform/srf06-cc26xx/contiki-conf.h
+++ b/platform/srf06-cc26xx/contiki-conf.h
@@ -262,7 +262,7 @@
  * @{
  */
 #ifndef ENERGEST_CONF_ON
-#define ENERGEST_CONF_ON            0 /**< Energest Module */
+#define ENERGEST_CONF_ON            1 /**< Energest Module */
 #endif
 /** @} */

